app/api/v1/presenters/root_presenter.rb

```diff
@@ -3,76 +3,16 @@ module Presenters
     module RootPresenter
       include Gris::Presenter
 
-      link :self do |opts|
-        "#{base_url(opts)}"
+      link :self do
+        Gris::Identity.base_url
       end
 
-      link :health do |opts|
-        { href: "#{base_url(opts)}/health" }
-      end
-
-      link :accounts do |opts|
-        {
-          href: "#{base_url(opts)}/accounts{?page,size}",
-          templated: true
-        }
-      end
-
-      link :account do |opts|
-        {
-          href: "#{base_url(opts)}/accounts/{id}",
-          templated: true
-        }
-      end
+      endpoint_link :health
 
-      link :features do |opts|
-        {
-          href: "#{base_url(opts)}/features{?page,size,account_id}",
-          templated: true
-        }
-      end
-
-      link :feature do |opts|
-        {
-          href: "#{base_url(opts)}/features/{id}",
-          templated: true
-        }
-      end
-
-      link :messages do |opts|
-        {
-          href: "#{base_url(opts)}/messages{?page,size,account_id}",
-          templated: true
-        }
-      end
-
-      link :message do |opts|
-        {
-          href: "#{base_url(opts)}/messages/{id}",
-          templated: true
-        }
-      end
-
-      link :routes do |opts|
-        {
-          href: "#{base_url(opts)}/routes{?page,size,account_id}",
-          templated: true
-        }
-      end
-
-      link :route do |opts|
-        {
-          href: "#{base_url(opts)}/routes/{id}",
-          templated: true
-        }
-      end
-
-      private
-
-      def base_url(opts)
-        request = Grape::Request.new(opts[:env])
-        request.base_url
-      end
+      resource_links :account
+      resource_links :feature, [:account_id]
+      resource_links :message, [:account_id]
```

Comment:
`account_id`. Its a replacement for the uri template options previously listed here (page and size come by default) - https://github.com/artsy/echo/pull/19/files#diff-d2263ae86a9a4d5fe8206564f28b5363L30